### ISSUE TITLE
Warning: Error in %>%: could not find function "%>%"

### ISSUE THREAD

**Issue by latlio** [rstudio/rsconnect#466]
When I try to deploy my ShinyApp, this error pops up is rsconnect::showLogs. 

I have checked the following:

- all libraries included (in particular, tidyverse, where %>% comes from)
- using namespace::function wherever possible
- no install.packages() calls anywhere
- no parsing errors (e.g. UTF-8 related)

Session info:
setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       macOS Catalina 10.15.7      
 system   x86_64, darwin17.0          
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Europe/London               
 date     2020-11-09

Package Dependencies:
             package   version source
1                   BH  1.72.0-3   CRAN
2                  DBI     1.1.0   CRAN
3                 MASS  7.3-51.6   CRAN
4               Matrix    1.2-18   CRAN
5                   R6     2.4.1   CRAN
6         RColorBrewer     1.1-2   CRAN
7                 Rcpp     1.0.5   CRAN
8            RcppEigen 0.3.3.7.0   CRAN
9         RcppParallel     5.0.2   CRAN
10         StanHeaders  2.21.0-6   CRAN
11             askpass       1.1   CRAN
12          assertthat     0.2.1   CRAN
13           backports    1.1.10   CRAN
14           base64enc     0.1-3   CRAN
15                blob     1.2.1   CRAN
16               broom     0.7.0   CRAN
17              bsplus     0.1.2   CRAN
18               callr     3.4.4   CRAN
19          cellranger     1.1.0   CRAN
20                 cli     2.0.2   CRAN
21               clipr     0.7.0   CRAN
22          colorspace     1.4-1   CRAN
23          commonmark       1.7   CRAN
24               cpp11     0.2.1   CRAN
25              crayon     1.3.4   CRAN
26           crosstalk   1.1.0.1   CRAN
27                curl       4.3   CRAN
28          data.table    1.13.0   CRAN
29              dbplyr     1.4.4   CRAN
30                desc     1.2.0   CRAN
31              digest    0.6.25   CRAN
32               dplyr     1.0.2   CRAN
33            ellipsis     0.3.1   CRAN
34            evaluate      0.14   CRAN
35               fansi     0.4.1   CRAN
36              farver     2.0.3   CRAN
37             fastmap     1.0.1   CRAN
38             forcats     0.5.0   CRAN
39                  fs     1.5.0   CRAN
40            generics     0.0.2   CRAN
41             ggplot2     3.3.2   CRAN
42                glue     1.4.2   CRAN
43              gtable     0.3.0   CRAN
44               haven     2.3.1   CRAN
45              hexbin    1.28.1   CRAN
46               highr       0.8   CRAN
47                 hms     0.5.3   CRAN
48           htmltools     0.5.0   CRAN
49         htmlwidgets     1.5.1   CRAN
50              httpuv     1.5.4   CRAN
51                httr     1.4.2   CRAN
52             isoband     0.2.2   CRAN
53            jsonlite     1.7.1   CRAN
54               knitr      1.29   CRAN
55            labeling       0.3   CRAN
56               later   1.1.0.1   CRAN
57             lattice   0.20-41   CRAN
58            lazyeval     0.2.2   CRAN
59           lifecycle     0.2.0   CRAN
60           lubridate     1.7.9   CRAN
61            magrittr       1.5   CRAN
62            markdown       1.1   CRAN
63                mgcv    1.8-31   CRAN
64                mime       0.9   CRAN
65              modelr     0.1.8   CRAN
66             munsell     0.5.0   CRAN
67                nlme   3.1-148   CRAN
68           onlineFDR     1.7.1 github
69             openssl     1.4.3   CRAN
70              pillar     1.4.6   CRAN
71            pkgbuild     1.1.0   CRAN
72           pkgconfig     2.0.3   CRAN
73             pkgload     1.1.0   CRAN
74              plotly   4.9.2.1   CRAN
75              praise     1.0.0   CRAN
76         prettyunits     1.1.1   CRAN
77            processx     3.4.4   CRAN
78            progress     1.2.2   CRAN
79            promises     1.1.1   CRAN
80                  ps     1.3.4   CRAN
81               purrr     0.3.4   CRAN
82               readr     1.3.1   CRAN
83              readxl     1.3.1   CRAN
84             rematch     1.0.1   CRAN
85              reprex     0.3.0   CRAN
86               rlang     0.4.7   CRAN
87           rmarkdown       2.3   CRAN
88           rprojroot     1.3-2   CRAN
89          rstudioapi      0.11   CRAN
90               rvest     0.3.6   CRAN
91              scales     1.1.1   CRAN
92             selectr     0.4-2   CRAN
93               sever     0.0.4   CRAN
94               shiny     1.5.0   CRAN
95             shinyBS      0.61   CRAN
96       shinyFeedback     0.2.0   CRAN
97        shinyWidgets     0.5.3   CRAN
98     shinycssloaders     1.0.0   CRAN
99      shinydashboard     0.7.1   CRAN
100 shinydashboardPlus     0.7.5   CRAN
101            shinyjs     2.0.0   CRAN
102        sourcetools     0.1.7   CRAN
103            stringi     1.5.3   CRAN
104            stringr     1.4.0   CRAN
105                sys       3.4   CRAN
106           testthat     2.3.2   CRAN
107             tibble     3.0.3   CRAN
108              tidyr     1.1.2   CRAN
109         tidyselect     1.1.0   CRAN
110          tidyverse     1.3.0   CRAN
111            tinytex      0.25   CRAN
112              tippy     0.1.0 github
113               utf8     1.1.4   CRAN
114              vctrs     0.3.4   CRAN
115        viridisLite     0.3.0   CRAN
116             waiter     0.1.3   CRAN
117            whisker       0.4   CRAN
118              withr     2.2.0   CRAN
119               xfun      0.17   CRAN
120               xml2     1.3.2   CRAN
121             xtable     1.8-4   CRAN
122               yaml     2.2.1   CRAN

**Comment by latlio** [comment]
I resolved the issue by realizing that I was sourcing in my ui.R and server.R scripts in a script called "app.R" (which contained all my libraries). Once I renamed "app.R" to "global.R", the issue was resolved. However, I'd love to know why this change worked. Is it something to do with how shinyapps.io expects a file named "global.R"?

**Comment by aronatkins** [comment]
The Shiny `shiny::runApp` function supports a few different file patterns in your application directory, including a standalone `app.R` and a `server.R` with a `ui.R`. The `global.R` file is a way of loading objects into the global environment made available to the Shiny application.

Shiny looks for `server.R` before `app.R`, meaning your `app.R` was never loaded (ignored).
https://github.com/rstudio/shiny/blob/1f9e4929a6d46ca4b1337faed8273b4d0b21b75c/R/shinyapp.R#L126-L136

When you renamed `app.R` to `global.R`, Shiny started processing that file during its startup, modifying the global environment.
https://github.com/rstudio/shiny/blob/1f9e4929a6d46ca4b1337faed8273b4d0b21b75c/R/shinyapp.R#L232-L243

In short - when you deploy a Shiny application with both `server.R` and `app.R`, the `server.R` function is used to launch the application.
